src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh

```diff
@@ -105,7 +108,11 @@ __vsc_prompt_cmd_original() {
 	for ((i = 0; i < ${#ADDR[@]}; i++)); do
 		builtin eval ${ADDR[i]}
 	done
-	IFS=''
+	if [[ ${__vsc_original_ifs+set} ]]; then
+		IFS="$__vsc_original_ifs"
```

Comment:
I'd suggest also adding `unset __vsc_original_ifs` here. Otherwise the logic *next time* could be polluted by the behavior *this time*. (Or maybe you can use `local` to declare `__vsc_original_ifs` somehow, I'm not really sure how that works.)